src/Storages/Kafka/KafkaConfigLoader.cpp

```diff
@@ -340,6 +349,16 @@ void updateGlobalConfiguration(
 {
     loadFromConfig(kafka_config, params, KafkaConfigLoader::CONFIG_KAFKA_TAG);
 
+    auto kafka_settings = storage.getKafkaSettings();
+    if (!kafka_settings[KafkaSetting::kafka_security_protocol].value.empty())
+        kafka_config.set("security.protocol", kafka_settings[KafkaSetting::kafka_security_protocol]);
+    if (!kafka_settings[KafkaSetting::kafka_sasl_mechanism].value.empty())
+        kafka_config.set("sasl.mechanism", kafka_settings[KafkaSetting::kafka_sasl_mechanism]);
+    if (!kafka_settings[KafkaSetting::kafka_sasl_username].value.empty())
+        kafka_config.set("sasl.username", kafka_settings[KafkaSetting::kafka_sasl_username]);
+    if (!kafka_settings[KafkaSetting::kafka_sasl_password].value.empty())
+        kafka_config.set("sasl.password", kafka_settings[KafkaSetting::kafka_sasl_password]);
+
```

Comment:
I have mixed feelings about exposing `KafkaSettings` directly. The reason is we have the `{Load,Consumer,Producer}ConfigParams` class to not to depend in specific settings and kind of unify the different source of parameters (config, query, named collections, etc). So maybe we should pass these values also part of `LoadConfigParams`.

On the other hand, adding 4 new members to that struct that might not be used is kind of polluting the config.

So I don't know which approach is better, but wanted to point this out. Feel free to choose either one: keeping it as is or including them to `LoadConfigParams`.

Reply:
I think I'd like to keep it simple here. We can always improve it later on.